src/utils/router.js

```diff
@@ -0,0 +1,48 @@
+import { isPlainObject } from './object'
+
+export const isRouterLink = tag => tag !== 'a'
+
+export const computeTag = (props, parent) => {
+  return parent.$router && props.to && !props.disabled
+    ? parent.$nuxt
+      ? 'nuxt-link'
+      : 'router-link'
+    : 'a'
+}
+
+export const computeRel = ({ target, rel }) => {
+  if (target === '_blank' && rel === null) {
+    return 'noopener'
+  }
+  return rel || null
+}
+
+export const computeHref = ({ disabled, href, to }, tag = 'a') => {
+  // We've already checked the parent.$router in computeTag,
+  // so isRouterLink(tag) indicates a live router.
```

Comment:
These comments are directed at functional components (i.e. parent.$router, vs. this.$router)..

Will probably need to update the comments.